### PR TITLE
pcmanfm: build with gtk3

### PR DESCRIPTION
**Pull request by lluchs**
###### Motivation for this change

pcmanfm works with both GTK 2 and 3. As recommended by #18559, I set it to build with GTK 3 per default.

###### Things done

Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers.

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/pcmanfm/default.nix [hunk 1]
```diff
@@ -1,13 +1,21 @@
-{ stdenv, fetchurl, glib, gtk2, intltool, libfm, libX11, pango, pkgconfig }:
+{ stdenv, fetchurl, glib, intltool, libfm, libX11, pango, pkgconfig
+, withGtk3 ? true, gtk2, gtk3 }:
 
+let
+  libfm' = libfm.override { withGtk3 = withGtk3; };
```
Comment (Ma27) [inline pkgs/applications/misc/pcmanfm/default.nix]: I'd prefer using `inherit` as this sort of assignment is kind of awkward:

```nix
libfm' = libfm.override {
  inherit withGtk3;
};
```

**Comment by Ma27** [comment]
I like the idea of the change, will try to test it in the next time :-)

**Comment by Mic92** [comment]
I have not looked into it deeply, but `lxpanel` fails to compile due it not find `libfm-gtk` in this pull request. I have not checked, if this was already the case before.

File: pkgs/applications/misc/pcmanfm/default.nix (hunk 1 above)
Comment (lluchs) [inline pkgs/applications/misc/pcmanfm/default.nix]: Ah yes, thanks for the hint.

**Comment by lluchs** [comment]
I changed the default on libfm back to gtk2 instead of gtk3. This way, it doesn't break any other gtk2 applications using the library.

**Comment by Ma27** [comment]
@lluchs I checked out your branch locally and built pcmanfm using `nix-build -A pcmanfm`.

It seems to work for me, so :+1: on these changes :-)

**Pull request merged**